packages/core/src/App/Components/Layout/Header/toggle-menu-drawer.jsx

```diff
@@ -14,163 +13,95 @@ import { observer, useStore } from '@deriv/stores';
+                        is_clickable
+                        lang={lang}
+                        toggleModal={() => {
+                            toggleDrawer();
+                            setMobileLanguageMenuOpen(false);
+                        }}
+                    />
+                ))}
+            </div>
+        </MobileDrawer.SubMenu>
+    )
+);
 
-const MenuLink = observer(
-    ({ link_to, icon, is_active, is_disabled, is_language, suffix_icon, text, onClickLink, is_hidden }) => {
-        const { common, ui, client } = useStore();
-        const { changeCurrentLanguage } = common;
-        const deriv_static_url = getStaticUrl(link_to);
-        const history = useHistory();
-        const { has_any_real_account, is_virtual } = client;
-        const { toggleReadyToDepositModal, toggleNeedRealAccountForCashierModal } = ui;
-        const real_account_needed_for_cashier = useIsRealAccountNeededForCashier();
-
-        const cashier_link =
-            link_to === routes.cashier_deposit ||
-            link_to === routes.cashier_withdrawal ||
-            link_to === routes.cashier_acc_transfer;
-
-        if (is_hidden) return null;
-
-        const traders_hub_path = window.location.pathname === routes.traders_hub;
-
-        if (real_account_needed_for_cashier && cashier_link && traders_hub_path) {
-            const handleClickCashier = () => {
-                onClickLink();
-                toggleNeedRealAccountForCashierModal();
-            };
-            return (
-                <div
-                    className={classNames('header__menu-mobile-link', {
-                        'header__menu-mobile-link--disabled': is_disabled,
-                    })}
-                    onClick={handleClickCashier}
-                >
-                    <Icon className='header__menu-mobile-link-icon' icon={icon} />
-                    <span className='header__menu-mobile-link-text'>{text}</span>
-                    {suffix_icon && <Icon className='header__menu-mobile-link-suffix-icon' icon={suffix_icon} />}
-                </div>
-            );
-        }
-
-        if (cashier_link && is_virtual && !has_any_real_account) {
-            const toggle_modal_routes =
-                window.location.pathname === routes.root || window.location.pathname === routes.traders_hub;
-
-            const toggleModal = () => {
-                if (toggle_modal_routes && !has_any_real_account) {
-                    toggleReadyToDepositModal();
-                }
-            };
+MobileLanguageMenu.displayName = 'MobileLanguageMenu';
 
-            const handleClickCashier = () => {
-                if (is_virtual && has_any_real_account) {
-                    history.push(routes.cashier_deposit);
-                } else if (!has_any_real_account && is_virtual) {
-                    toggleModal();
+const MenuTitle = React.memo(({ current_language, is_mobile_language_menu_open, setMobileLanguageMenuOpen }) => (
```

Comment:
@yauheni-deriv Same, Also this file is already too big, Can we move these into smaller components in separate files? 🙏🏻